lib/grpnames.gd

```diff
@@ -122,6 +142,21 @@
 ##   and where it is in the b-sequence if any. The the
 ##   linear algorithm above may be used.
 ##
+
+## The Kayal-Nezhmetdinov algorithm is described in
+## Neeraj Kayal and Timur Nezhmetdinov, Factoring Groups Efficiently, in
+## International Colloquium on Automata, Languages and Programming (ICALP) ,
+## Springer Verlag, 2009.
+# Gives a normal complement to the second argument if exists, fail otherwise.
+# In theory it finds the normal complement for infinite groups, as well,
+# but has an infinite loop if g/N is Abelian for infinite N.
+DeclareOperation( "ComplementNormalSubgroup", [IsGroup, IsGroup]);
+DeclareOperation( "ComplementNormalSubgroupNC", [IsGroup, IsGroup]);
```

Comment:
These two operations should probably be aligned with `ComplementClassesRepresentatives`... First off, in name: Perhaps `ComplementNormalSubgroupRepresentative`? Or perhaps the "normal" should be dropped?). Secondly, it would be nice if their comments referenced each other, so that people will find out about. Thirdly, I wonder if in certain cases, using `ComplementClassesRepresentatives` in the first place may be more efficient (I am not saying it is, and considering that `ComplementClassesRepresentatives` seems to compute more things, I'd not be surprised if it is always slower; but I wonder if you at least tested this?)